### PR TITLE
Implementing column renaming command

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by hishamco** [comment]
What about non SQL Server providers?!!

**Comment by sebastienros** [comment]
They use the base implementation. But apparently I made a mistake with mysql. Anyhow they will all work. Sqlite is the only one that only recently started supporting this.

**Comment by hishamco** [comment]
> Sqlite is the only one that only recently started supporting this.

Ya, I tried that few years ago without luck :( due the limitation

File: src/YesSql.Core/Sql/BaseComandInterpreter.cs [hunk 1]
```diff
@@ -193,6 +200,15 @@ public virtual void Run(StringBuilder builder, IAlterColumnCommand command)
             }
         }
 
+        public virtual void Run(StringBuilder builder, IRenameColumnCommand command)
+        {
+            builder.AppendFormat("alter table {0} rename column {1} to {2}",
```
Comment (hishamco) [inline src/YesSql.Core/Sql/BaseComandInterpreter.cs]:
```csharp
        builder.AppendFormat("alter table {0} rename column {1} to {2}",
            _dialect.QuoteForTableName(command.Name),
            _dialect.QuoteForColumnName(command.ColumnName),
            _dialect.QuoteForColumnName(command.NewColumnName)
        );
```

File: src/YesSql.Core/Sql/Schema/AlterTableCommand.cs [hunk 2]
```diff
@@ -46,6 +46,14 @@ public void AlterColumn(string columnName, Action<IAlterColumnCommand> column =
...
         }
 
+        public void RenameColumn(string columnName, string newColumnName)
+        {
+            var command = new RenameColumnCommand(Name, columnName, newColumnName);
+
+            TableCommands.Add(command);
+        }
+
+
```
Comment (hishamco) [inline src/YesSql.Core/Sql/Schema/AlterTableCommand.cs]: Remove extra line

File: src/YesSql.Provider.SqlServer/SqlServerComandInterpreter.cs [hunk 3]
```diff
@@ -7,5 +9,13 @@ public class SqlServerCommandInterpreter : BaseCommandInterpreter
         public SqlServerCommandInterpreter(ISqlDialect dialect) : base(dialect)
         {
         }
+
+        public override void Run(StringBuilder builder, IRenameColumnCommand command)
+        {
+            builder.AppendFormat("EXEC sp_RENAME {0}, {1}, 'COLUMN'",
+                _dialect.GetSqlValue(command.Name + "." + command.ColumnName),
+                _dialect.GetSqlValue(command.NewColumnName)
+                );
```
Comment (hishamco) [inline src/YesSql.Provider.SqlServer/SqlServerComandInterpreter.cs]: Same here

File: test/YesSql.Tests/CoreTests.cs [hunk 4]
```diff
@@ -3664,5 +3664,97 @@ public async Task ShouldCountSingleDocument()
...
+
+            using (var connection = _store.Configuration.ConnectionFactory.CreateConnection())
+            {
+                connection.Open();
+
+                try
+                {
+                    using (var transaction = connection.BeginTransaction(_store.Configuration.IsolationLevel))
+                    {
+
```
Comment (hishamco) [inline test/YesSql.Tests/CoreTests.cs]: Extra line

File: test/YesSql.Tests/CoreTests.cs [hunk 5]
```diff
@@ -3664,5 +3664,97 @@ public async Task ShouldCountSingleDocument()
...
+                    }
+                }
+                catch
+                {
+                    // Do nothing if the table can't be dropped
+                }
+
+                using (var transaction = connection.BeginTransaction(_store.Configuration.IsolationLevel))
+                {
+
```
Comment (hishamco) [inline test/YesSql.Tests/CoreTests.cs]: You can remove the extra lines everywhere

**Pull request merged**